### ISSUE TITLE
Weekly Calendar: Incorrect Fiscal Quarter

### ISSUE THREAD

**Issue by samaguire** [sql-bi/Bravo#478]
### Describe the problem

For the **Weekly** calendars the **Fiscal Quarter** column is a duplication of the Fiscal Year column.

### Bravo version

0.9.5 (0.9.8190.17886)

### Tabular model

Power BI Desktop, import mode

### Power BI

Power BI Desktop Version: 2.108.997.0 64-bit (August 2022)

### Steps to reproduce the behavior

Create a weekly calendar of any variant.

### Additional context

A fix would be moving the Fiscal Quarter to step 5 and update the caculation to `VAR __FiscalQuarter = "FQ" & __FiscalQuarterNumber`

``` DAX
VAR __Step3 = 
    GENERATE (
        __Calendar,
        VAR __LastTransactionDate = TODAY()
        VAR __FiscalYearNumber = [@YearNumber]
        VAR __FiscalYear = FORMAT ( __FiscalYearNumber, "\F\Y 0000" )
        VAR __FiscalQuarter = FORMAT ( __FiscalYearNumber, "\F\Y 0000" )
        RETURN ROW ( 
            "Fiscal Year Number", __FiscalYearNumber,
            "Fiscal Year", __FiscalYear,
            "Fiscal Quarter", __FiscalQuarter 
        )
    )
VAR __Step5 = 
    GENERATE (
        __Step3,
        VAR __LastTransactionDate = TODAY()
        VAR __FiscalYearNumber = [@YearNumber]
        VAR __Date = [Date]
        VAR __FiscalDayOfYearNumber = INT ( __Date - [@StartOfYear] + 1 )
        VAR __FiscalWeekNumber = INT ( CEILING ( __FiscalDayOfYearNumber / 7, 1 ) )
        VAR __FiscalQuarterNumber = IF ( __FiscalWeekNumber > 52, 4, ROUNDUP ( __FiscalWeekNumber / 13, 0 ) )
        VAR __FiscalYearQuarterNumber = __FiscalYearNumber * 4 - 1 + __FiscalQuarterNumber
        VAR __FiscalYearQuarter = "FQ" & __FiscalQuarterNumber & "-" & __FiscalYearNumber
        VAR __FiscalWeekInQuarterNumber = IF ( __FiscalWeekNumber > 52, 14, __FiscalWeekNumber - 13 * ( __FiscalQuarterNumber - 1 ) )
        VAR __FiscalMonthNumber = 
            ( __FiscalQuarterNumber - 1 ) * 3
                + SWITCH (
                    TRUE,
                    __FiscalWeekInQuarterNumber <= __WeeksInP1, 1,
                    __FiscalWeekInQuarterNumber <= ( __WeeksInP1 + __WeeksInP2 ), 2,
                    3
                )
        VAR __FiscalYearMonthNumber = __FiscalYearNumber * 12 - 1 + __FiscalMonthNumber
        VAR __FiscalWeek = FORMAT ( __FiscalWeekNumber, "\F\W00" )
        VAR __FiscalYearWeekNumber = INT ( DIVIDE ( __Date - __FirstWeekReference, 7 ) ) + 1
        VAR __FiscalYearWeek = FORMAT ( __FiscalWeekNumber, "\F\W00" ) & "-" & __FiscalYearNumber
        VAR __WeekDayNumber = WEEKDAY ( __Date, __WeekDayCalculationType )
        VAR __WeekDay = FORMAT ( __Date, "ddd" )
        RETURN ROW ( 
            "Sequential Day Number", INT ( __Date ),
            "Fiscal Quarter Number", __FiscalQuarterNumber,
            "Fiscal Year Quarter Number", __FiscalYearQuarterNumber,
            "Fiscal Year Quarter", __FiscalYearQuarter,
            "Fiscal Month Number", __FiscalMonthNumber,
            "Fiscal Year Month Number", __FiscalYearMonthNumber,
            "Fiscal Week Number", __FiscalWeekNumber,
            "Fiscal Week", __FiscalWeek,
            "Fiscal Year Week Number", __FiscalYearWeekNumber,
            "Fiscal Year Week", __FiscalYearWeek,
            "Day of Week Number", __WeekDayNumber,
            "Day of Week", __WeekDay,
            "Day of Fiscal Year Number", __FiscalDayOfYearNumber,
            "DateWithTransactions", __Date <= __LastTransactionDate 
        )
    )
```

becomes

``` DAX
VAR __Step3 = 
    GENERATE (
        __Calendar,
        VAR __LastTransactionDate = TODAY()
        VAR __FiscalYearNumber = [@YearNumber]
        VAR __FiscalYear = FORMAT ( __FiscalYearNumber, "\F\Y 0000" )
        RETURN ROW ( 
            "Fiscal Year Number", __FiscalYearNumber,
            "Fiscal Year", __FiscalYear
        )
    )
VAR __Step5 = 
    GENERATE (
        __Step3,
        VAR __LastTransactionDate = TODAY()
        VAR __FiscalYearNumber = [@YearNumber]
        VAR __Date = [Date]
        VAR __FiscalDayOfYearNumber = INT ( __Date - [@StartOfYear] + 1 )
        VAR __FiscalWeekNumber = INT ( CEILING ( __FiscalDayOfYearNumber / 7, 1 ) )
        VAR __FiscalQuarterNumber = IF ( __FiscalWeekNumber > 52, 4, ROUNDUP ( __FiscalWeekNumber / 13, 0 ) )
        VAR __FiscalQuarter = "FQ" & __FiscalQuarterNumber
        VAR __FiscalYearQuarterNumber = __FiscalYearNumber * 4 - 1 + __FiscalQuarterNumber
        VAR __FiscalYearQuarter = "FQ" & __FiscalQuarterNumber & "-" & __FiscalYearNumber
        VAR __FiscalWeekInQuarterNumber = IF ( __FiscalWeekNumber > 52, 14, __FiscalWeekNumber - 13 * ( __FiscalQuarterNumber - 1 ) )
        VAR __FiscalMonthNumber = 
            ( __FiscalQuarterNumber - 1 ) * 3
                + SWITCH (
                    TRUE,
                    __FiscalWeekInQuarterNumber <= __WeeksInP1, 1,
                    __FiscalWeekInQuarterNumber <= ( __WeeksInP1 + __WeeksInP2 ), 2,
                    3
                )
        VAR __FiscalYearMonthNumber = __FiscalYearNumber * 12 - 1 + __FiscalMonthNumber
        VAR __FiscalWeek = FORMAT ( __FiscalWeekNumber, "\F\W00" )
        VAR __FiscalYearWeekNumber = INT ( DIVIDE ( __Date - __FirstWeekReference, 7 ) ) + 1
        VAR __FiscalYearWeek = FORMAT ( __FiscalWeekNumber, "\F\W00" ) & "-" & __FiscalYearNumber
        VAR __WeekDayNumber = WEEKDAY ( __Date, __WeekDayCalculationType )
        VAR __WeekDay = FORMAT ( __Date, "ddd" )
        RETURN ROW ( 
            "Sequential Day Number", INT ( __Date ),
            "Fiscal Quarter Number", __FiscalQuarterNumber,
            "Fiscal Quarter", __FiscalQuarter ,
            "Fiscal Year Quarter Number", __FiscalYearQuarterNumber,
            "Fiscal Year Quarter", __FiscalYearQuarter,
            "Fiscal Month Number", __FiscalMonthNumber,
            "Fiscal Year Month Number", __FiscalYearMonthNumber,
            "Fiscal Week Number", __FiscalWeekNumber,
            "Fiscal Week", __FiscalWeek,
            "Fiscal Year Week Number", __FiscalYearWeekNumber,
            "Fiscal Year Week", __FiscalYearWeek,
            "Day of Week Number", __WeekDayNumber,
            "Day of Week", __WeekDay,
            "Day of Fiscal Year Number", __FiscalDayOfYearNumber,
            "DateWithTransactions", __Date <= __LastTransactionDate 
        )
    )
```

### Screenshots

![image](https://user-images.githubusercontent.com/62320770/189257474-9036dedc-9b0c-4462-91c4-6cb162ea9780.png)

**Comment by albertospelta** [comment]
@samaguire Thanks for reporting! This fix will be available in next release

**Comment by albertospelta** [comment]
Fixed in version [0.9.6](https://github.com/sql-bi/Bravo/releases/tag/v0.9.6)
